bindings/java/src/org/sleuthkit/datamodel/OsAccount.java

```diff
@@ -0,0 +1,549 @@
+/*
+ * Sleuth Kit Data Model
+ *
+ * Copyright 2020-2021 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Abstracts an OS user account.
+ *
+ * An OS user account may own files and (some) artifacts.
+ *
+ */
+public final class OsAccount implements Content {
+
+	final static long NO_USER = -1;
+	final static String NULL_UID_STR = null;
+
+	private final SleuthkitCase sleuthkitCase;
+	
+	private final long objId;	
+	private final OsAccountRealm realm;		// realm where the username is unique - a domain or a host name, may be null
+	private final String loginName;	// user login name - may be null
+	private final String uniqueId;	// a unique user sid/uid, may be null
+	private final String signature; // something that uniquely identifies this user - either the uniqueId or the realmName/userName.
+
+	private String fullName;	// full name
+	private boolean isAdmin = false;	// is admin account.
+	private OsAccountType osAccountType = OsAccountType.UNKNOWN;
+	private OsAccountStatus osAccountStatus = OsAccountStatus.UNKNOWN;
+	private long creationTime = 0;
+
+	private final List<OsAccountAttribute> osAccountAttributes = new ArrayList<>();
+
+	/**
+	 * Encapsulates status of an account - whether is it active or disabled or
+	 * deleted.
+	 */
+	public enum OsAccountStatus {
+		UNKNOWN(0, "Unknown"),
+		ACTIVE(1, "Active"),
+		DISABLED(2, "Disabled"),
+		DELETED(2, "Deleted");
```

Comment:
The account status is not part of the OsAccount constructor.  

@bcarrier  Should it be? Or perhaps Active should be assumed when the account is first created?

I had defined the Unknown status so that it could be initialized to Unknown when the object is first created.

Reply:
Removed status value Unknown.  Added status to the OsAccount constructor.  

When an account is first created, the status is assumed to be Active and written such in the database.

Reply:
Upon further conversation with Brian - the UNKNOWN status is being retained.  When an account is first created, its status will be set to UNKNOWN.  

The rationale is that when we create accounts from owner SIDs, we don't really know if the account is ACTIVE or not.